Source/WebCore/editing/Editor.cpp

```diff
@@ -3670,6 +3670,8 @@ RefPtr<TextPlaceholderElement> Editor::insertTextPlaceholder(const IntSize& size
     if (!placeholder->parentNode())
         return nullptr;
 
+    document->selection().setCaretVisibility(CaretVisibility::Hidden, { CaretVisibilitySuppressionReason::TextPlaceholderIsShowing });
```

Comment:
I was thinking more along the lines of adding two methods to `FrameSelection`, like `addCaretVisibilitySuppressionReason` and `removeCaretVisibilitySuppressionReason`. And then `FrameSelection` decides whether to show/hide the caret.

The current code is still a little confusing to read.